custom_components/google_home/sensor.py

```diff
@@ -29,96 +18,86 @@
-                            device.device_name,
-                            getattr(device, LABEL_ALARMS),
-                        ),
-                        GoogleHomeNextAlarmSensor(
-                            coordinator,
-                            entry,
-                            device.device_name,
-                            getattr(device, LABEL_ALARMS),
-                        ),
-                        GoogleHomeTimerSensor(
-                            coordinator,
-                            entry,
-                            device.device_name,
-                            getattr(device, LABEL_TIMERS),
-                        ),
-                        GoogleHomeNextTimerSensor(
-                            coordinator,
-                            entry,
-                            device.device_name,
-                            getattr(device, LABEL_TIMERS),
-                        ),
-                        GoogleHomeTokenSensor(
-                            coordinator,
-                            entry,
-                            device.device_name,
-                            device.local_auth_token,
-                        ),
-                    ]
-                )
-            else:
-                _LOGGER.warning(
-                    (
-                        "The %s device(hardware='%s') is not Google Home "
-                        "compatible and has no alarms/timers"
-                    ),
-                    device.device_name,
-                    device.hardware,
-                )
+        sensors.append(
+            GoogleHomeDeviceSensor(
+                coordinator,
+                entry,
+                device.name,
+                device.token,
+            )
+        )
+        if device.token and device.available:
+            sensors += [
+                GoogleHomeAlarmSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeNextAlarmSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeTimerSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeNextTimerSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+            ]
+    async_add_devices(sensors)
 
 
 class GoogleHomeSensorMixin:
     def get_device(self):
         """Return the device matched by name
         from the list of google devices in coordinator_data"""
         return next(
-            (
-                device
-                for device in self.coordinator.data
-                if device.device_name == self._name
-            ),
+            (device for device in self.coordinator.data if device.name == self._name),
```

Comment:
This should be a list.
```suggestion
            [device for device in self.coordinator.data if device.name == self._name],
```

Reply:
Can iterator not be a tuple? 🤔

Reply:
Nah, getting an error here if i change it to `list`:
```bash
    return next(
TypeError: 'list' object is not an iterator
```

Reply:
Yeah, actually tuple generator is the right way.